src/Psl/Xml/Issue/IssueCollection.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Xml\Issue;
+
+use Countable;
+use IteratorAggregate;
+use Psl\Arr;
+use Psl\Iter;
+use Psl\Iter\Iterator;
+use Psl\Str;
+
+final class IssueCollection implements Countable, IteratorAggregate
+{
+    /**
+     * @var list<Issue>
+     */
+    private array $issues;
+
+    public function __construct(Issue ...$errors)
+    {
+        $this->issues = $errors;
+    }
+
+    /**
+     * @return Iterator<int, Issue>
+     */
+    public function getIterator(): Iterator
+    {
+        return Iterator::create($this->issues);
+    }
+
+    public function count(): int
+    {
+        return count($this->issues);
+    }
+
+    /**
+     * @param (callable(Issue): bool) $filter
+     */
+    public function filter(callable $filter): self
+    {
+        return new self(
+            ...Arr\filter(
+                $this->issues,
+                $filter
+            )
+        );
+    }
+
+    /**
+     * @psalm-suppress InvalidNullableReturnType
+     */
+    public function getHighestLevel(): Level
+    {
+        /** @psalm-suppress NullableReturnStatement */
```

Comment:
Looks like the reduce typing doesn't work that well...
It thinks that the first argument passed to reducer can be null, but clearly it never is.
Is this something we need to fix in the reduce function? Or do I need to add something to the reducer's function docblocks in order to let psalm know it is never null?

Reply:
hm, not sure, but we should use Math\max_by here.